### ISSUE TITLE
Rework OpenVPN configuration frontend

### ISSUE THREAD

**Issue by simonszu** [opnsense/core#3146]
This issue happens on OPNsense 18.7.10-amd64.
The URL to the web config where to replicate this issue is: /vpn_openvpn_server.php

Description of the issue: 
I want to configure a very simple OpenVPN peer2peer connection with PSK. In OpenVPN config format this configuration looks like this:
```
dev tun3
proto udp
ifconfig 172.23.211.129 172.23.211.1
comp-lzo
port 42003
secret psk.key
log-append /var/log/openvpn/p2p.log
verb 4
persist-tun
persist-key
mode p2p
```
The crucial part is the `ifconfig` line. It specifies which address my tunnel endpoint should have (`.129`) and which address the remote endpoint should have (`.1`). The other endpoint has this setting switched. Both endpoints expect the other site to have exactly the IP address which is set here. If the other site has another address, the connection does not work. 

Currently, the OpenVPN server setting for P2P PSK do not allow this kind of configuration. I have already investigated that the `ifconfig` line in the generated config is directly affected by the "IPv4 Tunnel Network" setting, which causes the config file writer to select two random IP addresses from this setting to build the `ifconfig` line. This may work in a scenario where the Opnsense administrator has full configuration power over all VPN peers which participate in this P2P setup. 

However, it does not work in a scenario where the Opnsense administrator only has power over the Opnsense instance, and has to fulfill some assumptions given by the other peer(s). It currently is simply not possible to explicitly specify this `ifconfig` line. 

A possible solution i have already tried was to enter the `ifconfig` settings manually in the "Advanced settings" textbox, however, the web configurator does not allow this, since "IPv4 Tunnel Network" is a mandatory setting. 

Summarized, i do not know if this issue is a bug (since Opnsense does not allow the simplest OpenVPN configuration which can exist) or a feature (like the ability to manually specify the `ifconfig` setting).

As a simple possible solution, i would suggest to just make the "IPv4 Tunnel Network" setting optional, and to let the users manually specify the IP settings in the advanced settings textbox. 

As a more advanced solution, i would suggest to add more modes to the "Server modes" dropdown box. The current mode "Peer to Peer (Shared Key)" is to rename like "Peer to Peer with Tunnel Network (Shared Key)" and left untouched. A second mode like "Peer to Peer Direct connection" is added, where all IP settings text boxes are removed and only two text boxes are added, like "IPv4 local tunnel IP" and "IPv4 remote tunnel IP", and similar fpor IPv6. 

As a _really_ advanced solution, the differentiation between "Servers" and "Clients" in the left sidebar does not really fit a Peer to Peer connection. Yes, there technically is a "server" and a "client" in p2p mode, but the only difference is that the "client" has to specify a "remote". 
So, the possibly best solution would be to add a third option to the sidebar, next to "Servers" and "Clients", called "P2P". There, all necessary P2P PSK settings can be entered. It has a "remote" field which is fully optional, and the `remote` setting is only written to the "real" log file when the remote field has content.

**Comment by johnr14** [comment]
I opened a issue with openvpn front end also : https://github.com/opnsense/plugins/issues/1199

If a rework of the front end is required, I'd also mention the following needs : 
+ **Get public ip** from VPN and assign it to the interface (+recheck if changed), display it in **connection status** and on **dashboard**.
  _Using a online service like ifconfig.co or from vpn provider_
+ **Get port forwarding** from VPN and **update firewall rule** automatically (to destination IP:port)
  _Allow running of a script or hard code a few VPN provider
   Example is [PIA](https://www.privateinternetaccess.com/helpdesk/kb/articles/can-i-use-port-forwarding-without-using-the-pia-client) and the need to run a script to return the port_ 
+ **Enable/disable firewall rules** based on the presence or absence of vpn connection
+ **Add checkbox** for additional options : 
  _persist-key
  persist-tun
  tls-client
  remote-cert-tls server
  auth-retry interact_
+ **Add warning icon** beside connection if encryption and digest algorithm is different, as it is just logged.
   _Aug 10 18:22:44 | openvpn[76289]: WARNING: 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128'
-- | --
   Aug 10 18:22:44 | openvpn[76289]: WARNING: 'auth' is used inconsistently, local='auth SHA256', remote='auth SHA1'
   Aug 10 18:22:44 | openvpn[76289]: WARNING: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher BF-CBC'_
+ **Add connectivity test server** (for ping/curl)
  _Check if connection is ok by communicating with a computer at the other end. If no connectivity, change to an other vpn server_

Thanks

**Comment by dkowis** [comment]
I have worked around this by using the "Advanced Configuration" field that's "going to go away eventually". I went looking for a way to request that it *not* go away, so that things like this could be accomplished easily :) There's a lot of advanced configuration for OpenVPN, and it's handy to be able to write proper configuration, when you know what you're doing.

I don't agree with the help text that tells you it's "Inherently insecure." Having configured many OpenVPNs by hand on linux in a config file, it's really just a power-user interface :)

I'm cool with a UI that's advanced and capable, but it's awful difficult to cover ever edge case imaginable. There's always use in having an "advanced config" that lets you write the configuration yourself. Here there be dragons disclaimer is fine :)
